### ISSUE TITLE
error in trying to build opsdroid docker image

### ISSUE THREAD

**Issue by bparbhu** [opsdroid/opsdroid#1874]
Hi guys,

So I tried building the opsdroid image from scratch by pulling down changes from the new release on the master branch,
then cd into the opsdroid directory and use the docker build command to build the image.

When I initiate the build I get this error message which stops it from happening.

```
ERROR: Could not find a version that satisfies the requirement pycryptodome<4.0.0,>=3.10.1 (from matrix-nio) (from versions: none)
ERROR: No matching distribution found for pycryptodome<4.0.0,>=3.10.1
The command '/bin/sh -c apk add --no-cache git olm libzmq && pip install --no-cache-dir --no-index -f ${DEPS_DIR} $(find ${DEPS_DIR} -type f -name opsdroid--any.whl)${EXTRAS} && rm -rf /tmp/ /var/tmp/* ${DEPS_DIR}/* && adduser -u 1001 -S -G root opsdroid' returned a non-zero code: 1
```

I also reached out to the opsdroid community on element and had this error reproduced by @Sleuth56 

His process was as followed 

```
git clone https://github.com/opsdroid/opsdroid/
cd opsdroid
docker build .
```

and the resulting error

```
INFO: pip is looking at multiple versions of opsdroid[all,connector-matrix-e2e] to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement pycryptodome<4.0.0,>=3.10.1 (from matrix-nio) (from versions: none)
ERROR: No matching distribution found for pycryptodome<4.0.0,>=3.10.1
The command '/bin/sh -c apk add --no-cache     git     olm     libzmq     && pip install --no-cache-dir --no-index -f ${DEPS_DIR}     $(find ${DEPS_DIR} -type f -name opsdroid-*-any.whl)${EXTRAS}     && rm -rf /tmp/* /var/tmp/* ${DEPS_DIR}/*     && adduser -u 1001 -S -G root opsdroid' returned a non-zero code: 1
```

Also @Sleuth56 wanted to link https://github.com/opsdroid/opsdroid/runs/4451560556
to this issue as well

Just want to see if this is some sort of pip related problem but just wanted to bring this to everyone's attention.

Thanks again and always appreciative of the work you guys are doing,

-Brian

**Comment by bparbhu** [comment]
Hey All,

Just wanted to follow up with this and see where things are at.

Thanks again and always appreciative,

-Brian

**Comment by Sleuth56** [comment]
I bugged the devs the other day and they said that they are really busy right now. They are aware of this issue.

**Comment by vinayman** [comment]
Hi - nice to meet you all.

I saw this project on up-for-grabs.net so thought I could just look at any issues and help out whatever is open - hope that's okay!

I've raised a PR above which resolves the issue - to summarise:

The issue is that the docker build pulls down the pycryptodome dependency as a zip file rather than a whl or tar.gz (where tar.gz is correctly converted to a whl file).
The simple fix is to convert the zip to a whl file so it is installed during the later stage of the build

Thanks!
